Add documentation for manual testing (#38)
undefinedlabs/scope-docs: docs/ios-manual-testing.md

title: Scope iOS Agent Manual testing
sidebar_label: Manual testing
---

The Scope iOS agent allows you to perform manual tests on your application that will be recorded for later troubleshooting in Scope. You can either run your application from Xcode in your simulator or device, or distribute a build to your testers with the functionality built-in.

The results will appear in your **Local Development scratchpad** in Scope

## Installation

Link your application target with `ScopeAgent`

<!--DOCUSAURUS_CODE_TABS-->
<!--Cocoapods-->
By adding the pod to your `Podfile` and running `pod install`. For example:

```
target 'MyApp' do
 pod 'ScopeAgent'
end
```

or

```
target 'MyFramework' do
 pod 'ScopeAgent'
end
```

<!--Carthage-->
Add the `ScopeAgent` dependency to your `Cartfile` if not already done in previous steps, and run `cart update`

```
binary "https://releases.undefinedlabs.com/scope/agents/ios/ScopeAgent.json"
```

In your application or framework targets, add `ScopeAgent.framework` located in `Carthage/Build/iOS` to the
*Linked frameworks and Libraries* in General target settings or to the *Link Binaries With Libraries* build phase. 

<!--END_DOCUSAURUS_CODE_TABS-->

>  If you want to run your application from Xcode in your simulator or device, the following environment variables must be set in your **Run target** ([instructions](https://help.apple.com/xcode/mac/10.1/index.html?localePath=en.lproj#/dev3ec8a1cb4)):

Reviewer: ```suggestion
If you want to run your application from Xcode in your simulator or device, the following environment variables must be set in your **Run target** ([instructions](https://help.apple.com/xcode/mac/10.1/index.html?localePath=en.lproj#/dev3ec8a1cb4)):
```

| Key                        | Value                         |
| -------------------------- | ----------------------------- |
| `SCOPE_XCODE_APIKEY`       | `$(SCOPE_XCODE_APIKEY)`       |
| `SCOPE_XCODE_API_ENDPOINT` | `$(SCOPE_XCODE_API_ENDPOINT)` |

> Alternatively, if you want to distribute the instrumented application to you QA team, then you must add the following entries in your application Info.plist:

Reviewer: ```suggestion
Alternatively, if you want to distribute the instrumented application (e.g. to the QA team), then you must add the following entries in your application's `Info.plist`:
```

| Key                  | Type     | Value                         |
| -------------------- | -------- | ----------------------------- |

> | `SCOPE_APIKEY`       | `String` | `< your Scope APIKEY >`       |

Reviewer: ```suggestion
| `SCOPE_APIKEY`       | `String` | `<Your Scope API key>`        |
```

> | `SCOPE_API_ENDPOINT` | `String` | `< your Scope API_ENDPOINT >` |

Reviewer: ```suggestion
| `SCOPE_API_ENDPOINT` | `String` | `<Your Scope API endpoint>`   |
```

---
id: ios-manual-testing
title: Scope iOS Agent Manual testing
sidebar_label: Manual testing
---

> Scope allows to do manual test runs that automatically connect to your Scope instance. You can run your application from Xcode in your simulator or device, but you can also distribute a build to your testers with the functionality ready.

Reviewer: ```suggestion
The Scope iOS agent allows you to perform manual tests on your application that will be recorded for later troubleshooting in Scope. You can either run your application from Xcode in your simulator or device, or distribute a build to your testers with the functionality built-in.
```

> The results will appear in your local development scratchpad in Scope

Reviewer: ```suggestion
The results will appear in your **Local Development scratchpad** in Scope
```

## Installation

Link your application target with `ScopeAgent`

<!--DOCUSAURUS_CODE_TABS-->
<!--Cocoapods-->
By adding the pod to your `Podfile` and running `pod install`. For example:

```
target 'MyApp' do
 pod 'ScopeAgent'
end
```

or

```
target 'MyFramework' do
 pod 'ScopeAgent'
end
```

<!--Carthage-->
Add the `ScopeAgent` dependency to your `Cartfile` if not already done in previous steps, and run `cart update`

```
binary "https://releases.undefinedlabs.com/scope/agents/ios/ScopeAgent.json"
```

In your application or framework targets, add `ScopeAgent.framework` located in `Carthage/Build/iOS` to the
*Linked frameworks and Libraries* in General target settings or to the *Link Binaries With Libraries* build phase. 

<!--END_DOCUSAURUS_CODE_TABS-->

> Then you must add the following entries in your application Info.plist

Reviewer: This step depends on whether the user wants to hardcode the values for distribution or not. I would put first the instructions of the env vars for local Xcode runs, and later, as an optional step, update the `Info.plist` if the user wants to distribute the application with manual testing activated.

Author: Changed

| Key                  | Type     | Value                         |
| -------------------- | -------- | ----------------------------- |
| `SCOPE_APIKEY`       | `String` | `< your Scope APIKEY >`       |
| `SCOPE_API_ENDPOINT` | `String` | `< your Scope API_ENDPOINT >` |

You can distribute this app to your QA Team so they will report to the selected ENDPOINT

Alternatively, if you are going to run the app launching from Xcode in local development you can add the following environment variables in your **Run target** ([instructions](https://help.apple.com/xcode/mac/10.1/index.html?localePath=en.lproj#/dev3ec8a1cb4)):

| Key                        | Value                         |
| -------------------------- | ----------------------------- |
| `SCOPE_XCODE_APIKEY`       | `$(SCOPE_XCODE_APIKEY)`       |
| `SCOPE_XCODE_API_ENDPOINT` | `$(SCOPE_XCODE_API_ENDPOINT)` |

## Usage

> After the application launches, a Scope logo will appear in the screen over your application elements. When you want to start a test you must simple press, and it will allow to start a new test.

Reviewer: ```suggestion
After the application launches, a floating Scope logo will appear in the screen over your application elements. When you want to start a test, click on the logo and provide the test with a name.
```

>  While the test is running the Scope logo will be glowing in red, if you click the logo again it will allow to stop the recording and save the state of the test. 

Reviewer: ```suggestion
While the test is being recorded, the Scope logo will be glowing in red. To finish the test and stop the recording, click on the logo, and choose whether the test passed or failed. 
```

> After saving, the test will appear in Scope.

Reviewer: ```suggestion
Manual tests will appear in your **Local Development scratchpad** section.
```
